CRM/Contribute/BAO/Contribution.php

```diff
@@ -4041,7 +4035,8 @@ public static function getPaymentInfo($id, $component, $getTrxnInfo = FALSE, $us
       $total = $total['total_amount'];
     }
 
-    $paymentBalance = CRM_Core_BAO_FinancialTrxn::getPartialPaymentWithType($id, $entity, FALSE, $total);
+    $paymentBalance = CRM_Contribute_BAO_Contribution::getContributionBalance($contributionId, $total);
+
```

Comment:
@eileenmcnaughton this will cause an issue with the calculation done below(see L4049, L4057),  whereas it is expecting $paymentBalance to be an amount but as per the patch it always returns a array.

Reply:
@monishdeb - no getContributionBalance returns an amount - when it's called within getPartialPaymentByType they take that amount & create an array for presumably legacy reasons

Reply:
Oops sorry, my bad didn't saw what function is been called here. Hmm let me do a quick test at UI level, then I'll merge it.

Reply:
thanks @monishdeb- I think once https://github.com/civicrm/civicrm-core/pull/13610 is merged & we can clean up the payment form it's going to be easy to remove that getPartialPaymentWithType function altogether.

And now ... all Civi devs repeat after me

 "if a function doesn't return the format I want, I will never again just add another parameter to get it to return a different format. I understand a wrapper function is acceptable if I really need to change the format but only if I ensure  just butchering something unrelated onto the function" :-)

Reply:
Hmm rationally, its good in terms of clarity and the definite role each wrapper function has to play rather then bring more parameters to it and to extend its functionality, which is a pain for dev to understand the code (like the export and changeFee fns).  Bad in terms of increase in SLOC but thats fine, we have to be more descriptive on how we define a function. It really helps devs to understand the code or figure out the bug and bring any change without worrying about different scenarios.

Reply:
:-)

Reply:
SLOC = single line of code? I think drupal has some stuff around that in it's standards (basically goal is readability not brevity)

Reply:
Source LOC